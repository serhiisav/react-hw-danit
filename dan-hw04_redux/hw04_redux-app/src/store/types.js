export const ADD_ALL_ITEMS = 'ADD_ALL_ITEMS';
export const ADD_ITEMS_FAV_LIST = 'ADD_ITEMS_FAV_LIST';
export const ADD_ITEMS_CART_LIST = 'ADD_ITEMS_CART_LIST';
export const ADD_ITEMS_FAILURE = 'ADD_ITEMS_FAILURE';
export const ADD_ITEMS_STARTED = 'ADD_ITEMS_STARTED';
export const CHANGE_FAV_LIST = 'CHANGE_FAV_LIST';
export const CHANGE_CART_LIST = 'CHANGE_CART_LIST';
export const SET_FAV_LIST = 'SET_FAV_LIST';
export const SET_CART_LIST = 'SET_CART_LIST';
// export const SET_IS_FAVORITE = 'SET_IS_FAVORITE';
