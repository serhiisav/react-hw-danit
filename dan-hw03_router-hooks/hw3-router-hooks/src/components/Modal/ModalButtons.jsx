import React from "react";

const ModalButtons = ({ dataModal, setModalStatus, handleClickModalOk }) => {
    const { btnActionOk, btnActionCancel } = dataModal;

    return (
        <>
            <button onClick={handleClickModalOk} id='btnModalOk' className='btnModal' type='button'>{btnActionOk}</button>
            <button onClick={setModalStatus} id='btnModalCancel' className='btnModal' type='button'>{btnActionCancel}</button>
        </>
    )
}

export default ModalButtons;