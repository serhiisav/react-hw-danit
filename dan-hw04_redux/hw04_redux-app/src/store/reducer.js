import { initialState } from "./store"

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_ITEMS_STARTED': {
            return { ...state, isLoading: true }
        }
        case 'ADD_ALL_ITEMS': {
            return { ...state, products: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_FAV_LIST': {
            return { ...state, itemsFavoriteList: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_CART_LIST': {
            return { ...state, itemsCartList: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_FAILURE': {
            return { ...state, isLoading: false, error: action.payload }
        }
        case 'SET_FAV_LIST': {
            return { ...state, favoritesList: action.payload }
        }
        case 'SET_CART_LIST': {
            return { ...state, cartList: action.payload }
        }
        case 'CHANGE_FAV_LIST': {
            return {
                ...state,
                favoritesList: state.favoritesList.includes(action.payload) ?
                    state.favoritesList.filter(el => el !== action.payload) :
                    [...state.favoritesList, action.payload]
            }
        }
        case 'CHANGE_CART_LIST': {
            return {
                ...state,
                cartList: state.cartList.includes(action.payload) ?
                    state.cartList.filter(el => el !== action.payload) :
                    [...state.cartList, action.payload]
            }
        }
        case 'SET_IS_OPENED_MODAL': {
            return { ...state, isOpenedModal: action.payload }
        }
        case 'SET_DATA_MODAL': {
            return { ...state, dataModal: action.payload }
        }
        default: {
            return state;
        }
    }
}

export default reducer;