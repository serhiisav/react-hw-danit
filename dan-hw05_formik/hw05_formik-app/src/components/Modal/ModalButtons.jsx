import React from "react";
import { useSelector } from "react-redux";
import useOpenModal from "../../hooks/useOpenModal";

const ModalButtons = () => {
    const { btnActionOk, btnActionCancel, id } = useSelector(state => state.dataModal);
    const { setModalStatus, handleClickModalOk } = useOpenModal();

    return (
        <>
            <button onClick={() => handleClickModalOk(id)} id='btnModalOk' className='btnModal' type='button'>{btnActionOk}</button>
            <button onClick={setModalStatus} id='btnModalCancel' className='btnModal' type='button'>{btnActionCancel}</button>
        </>
    )
}

export default ModalButtons;