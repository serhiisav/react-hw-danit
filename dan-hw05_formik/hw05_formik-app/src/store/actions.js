import {
    ADD_ALL_ITEMS,
    ADD_ITEMS_FAV_LIST,
    ADD_ITEMS_CART_LIST,
    ADD_ITEMS_FAILURE,
    ADD_ITEMS_STARTED,
    CHANGE_FAV_LIST,
    CHANGE_CART_LIST,
    SET_FAV_LIST,
    SET_CART_LIST,
    REMOVE_CART_LIST,
    SET_IS_ACTIVE_ORDER_FORM,
    SET_ORDER_INFO,
    REMOVE_ITEMS_CART_LIST,
    SET_TOTALS_ORDER

} from './types'

export const addAllItemsSuccess = (items) => {
    return {
        type: ADD_ALL_ITEMS,
        payload: items
    }
}
export const addItemsFavList = (itemsFavoriteList) => {
    return {
        type: ADD_ITEMS_FAV_LIST,
        payload: itemsFavoriteList,
    }
}
export const addItemsCartList = (itemsCartList) => {
    return {
        type: ADD_ITEMS_CART_LIST,
        payload: itemsCartList,
    }
}
export const addItemsFailure = (error) => {
    return {
        type: ADD_ITEMS_FAILURE,
        payload: error
    }
}
export const addItemsStarted = () => {
    return {
        type: ADD_ITEMS_STARTED,
    }
}
export const setFavList = (starList) => {
    return {
        type: SET_FAV_LIST,
        payload: starList,
    }
}
export const setCartList = (cartList) => {
    return {
        type: SET_CART_LIST,
        payload: cartList,
    }
}
export const changeFavListAction = (id) => {
    return {
        type: CHANGE_FAV_LIST,
        payload: id,
    }
}
export const changeCartListAction = (id) => {
    return {
        type: CHANGE_CART_LIST,
        payload: id,
    }
}
export const removeCartList = () => {
    return {
        type: REMOVE_CART_LIST,
    }
}
export const removeItemsCartList = () => {
    return {
        type: REMOVE_ITEMS_CART_LIST,
    }
}
export const setIsActiveOrderForm = (isActive) => {
    return {
        type: SET_IS_ACTIVE_ORDER_FORM,
        payload: { isActive },
    }
}
export const setOrderInfo = (values) => {
    return {
        type: SET_ORDER_INFO,
        payload: { values },
    }
}
export const setTotalsOrder = (orderSum) => {
    return {
        type: SET_TOTALS_ORDER,
        payload: { orderSum },
    }
}