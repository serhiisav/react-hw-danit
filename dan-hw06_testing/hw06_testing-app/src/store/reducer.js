
export const initialState = {
    products: [],
    favoritesList: [],
    cartList: [],
    isOpenedModal: false,
    isLoading: false,
    error: null,
    itemsFavoriteList: [],
    itemsCartList: [],
    isActiveOrderForm: false,
    orderInfo: {},
    totalsOrder: null,
    dataModal: {
        header: 'Modal',
        closeButton: true,
        text: 'Are you sure you want to delete it',
        btnActionOk: 'OK',
        backgroundColor: '#b46c7b',
        btnActionCancel: 'CANCEL',
    }
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_ITEMS_STARTED': {
            return { ...state, isLoading: true }
        }
        case 'ADD_ALL_ITEMS': {
            return { ...state, products: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_FAV_LIST': {
            return { ...state, itemsFavoriteList: action.payload, isLoading: false, error: null }
        }
        case 'ADD_ITEMS_CART_LIST': {
            return { ...state, itemsCartList: action.payload, isLoading: false, error: null }
        }
        case 'REMOVE_ITEMS_CART_LIST': {
            return { ...state, itemsCartList: [] }
        }
        case 'REMOVE_CART_LIST': {
            return { ...state, cartList: [] }
        }
        case 'ADD_ITEMS_FAILURE': {
            return { ...state, isLoading: false, error: action.payload }
        }
        case 'SET_FAV_LIST': {
            return { ...state, favoritesList: action.payload }
        }
        case 'SET_CART_LIST': {
            return { ...state, cartList: action.payload }
        }
        case 'CHANGE_FAV_LIST': {
            return {
                ...state,
                favoritesList: state.favoritesList.includes(action.payload) ?
                    state.favoritesList.filter(el => el !== action.payload) :
                    [...state.favoritesList, action.payload]
            }
        }
        case 'CHANGE_CART_LIST': {
            return {
                ...state,
                cartList: state.cartList.includes(action.payload) ?
                    state.cartList.filter(el => el !== action.payload) :
                    [...state.cartList, action.payload]
            }
        }
        case 'SET_IS_OPENED_MODAL': {
            return { ...state, isOpenedModal: action.payload }
        }
        case 'SET_DATA_MODAL': {
            return { ...state, dataModal: action.payload }
        }
        case 'SET_IS_ACTIVE_ORDER_FORM': {
            return { ...state, isActiveOrderForm: action.payload.isActive }
        }
        case 'SET_ORDER_INFO': {
            return { ...state, orderInfo: { ...action.payload.values, totalSum: state.totalsOrder, orderList: state.itemsCartList } }
        }
        case 'SET_TOTALS_ORDER': {
            return { ...state, totalsOrder: action.payload.orderSum }
        }
        default: {
            return state;
        }
    }
}

export default reducer;