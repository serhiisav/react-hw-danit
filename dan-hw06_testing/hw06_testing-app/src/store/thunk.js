import { getItems } from '../api/getItems';
import { addAllItemsSuccess, addItemsFailure, addItemsStarted, changeCartListAction, changeFavListAction } from "./actions"


export const addAllItems = () => (dispatch, getState) => {
    const { products } = getState();
    if (products.length === 0) {
        dispatch(addItemsStarted());
        getItems()
            .then(response => dispatch(addAllItemsSuccess(response.data)))
            .catch(error => dispatch(addItemsFailure(error.message)));
    }
}

export const changeFavoritesList = (id) => (dispatch, getState) => {
    dispatch(changeFavListAction(id));
    const { favoritesList } = getState();
    localStorage.setItem('starList', JSON.stringify(favoritesList));
}

export const changeCartList = (id) => (dispatch, getState) => {
    dispatch(changeCartListAction(id));
    const { cartList } = getState();
    localStorage.setItem('cartList', JSON.stringify(cartList));
}