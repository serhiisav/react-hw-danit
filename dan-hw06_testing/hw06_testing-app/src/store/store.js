import { createStore, applyMiddleware } from 'redux';
import reducer from './reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

export const initialState = {
    products: [],
    favoritesList: [],
    cartList: [],
    isOpenedModal: false,
    isLoading: false,
    error: null,
    itemsFavoriteList: [],
    itemsCartList: [],
    isActiveOrderForm: false,
    orderInfo: {},
    totalsOrder: null,
    dataModal: {
        header: 'Modal',
        closeButton: true,
        text: 'Are you sure you want to delete it',
        btnActionOk: 'OK',
        backgroundColor: '#b46c7b',
        btnActionCancel: 'CANCEL',
    }
}
const store = createStore(
    reducer,
    // initialState,
    composeWithDevTools(applyMiddleware(thunk))
    // window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;