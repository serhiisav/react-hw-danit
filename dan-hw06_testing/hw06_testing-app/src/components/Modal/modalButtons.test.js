import userEvent from '@testing-library/user-event'
import { cleanup, render, screen } from '@testing-library/react';
import ModalButtons from './ModalButtons';
import { Provider } from 'react-redux';
import store from '../../store/store';
import reducer from '../../store/reducer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom/extend-expect';



describe('<ModalButtons/> component', () => {
    afterEach(() => cleanup());

    it('should render', () => {
        render(<Provider store={store}><ModalButtons></ModalButtons></Provider>);
        screen.getByRole('button', { name: /ok/i });
        screen.getByRole('button', { name: /cancel/i });
    });

    it('renders correctly', () => {
        const { container } = render(<Provider store={store}><ModalButtons></ModalButtons></Provider>);
        expect(container).toMatchSnapshot();
    });

    it('should trigger onClick functions button "OK"', () => {
        const initialState = {
            isOpenedModal: false,
            cartList: [],
            dataModal: {
                id: 2,
                btnActionOk: 'OK',
                btnActionCancel: 'CANCEL',
            }
        }
        const store = createStore(
            reducer,
            initialState,
            applyMiddleware(thunk));

        render(
            <Provider store={store}>
                <ModalButtons></ModalButtons>
            </Provider>
        );

        userEvent.click(screen.getByRole('button', { name: /ok/i }));

        expect(store.getState()).toEqual({
            isOpenedModal: true,
            cartList: [2],
            dataModal: {
                id: 2,
                btnActionOk: 'OK',
                btnActionCancel: 'CANCEL'
            }
        });
    });

    it('should trigger onClick functions button "CANCEL"', () => {
        const initialState = {
            isOpenedModal: true,
            dataModal: {
                btnActionOk: 'OK',
                btnActionCancel: 'CANCEL',
            }
        }
        const store = createStore(reducer, initialState);

        render(
            <Provider store={store}>
                <ModalButtons></ModalButtons>
            </Provider>
        );

        userEvent.click(screen.getByRole('button', { name: /cancel/i }));
        expect(store.getState().isOpenedModal).toBe(false);
    });
});