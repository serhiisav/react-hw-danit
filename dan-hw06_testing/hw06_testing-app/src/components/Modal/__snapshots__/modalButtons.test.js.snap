// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalButtons/> component renders correctly 1`] = `
<div>
  <button
    class="btnModal"
    id="btnModalOk"
    type="button"
  >
    OK
  </button>
  <button
    class="btnModal"
    id="btnModalCancel"
    type="button"
  >
    CANCEL
  </button>
</div>
`;

exports[`<ModalButtons/> component should render 1`] = `
<div>
  <button
    class="btnModal"
    id="btnModalOk"
    type="button"
  >
    OK
  </button>
  <button
    class="btnModal"
    id="btnModalCancel"
    type="button"
  >
    CANCEL
  </button>
</div>
`;
